psi4/src/psi4/libmints/basisset.cc

```diff
@@ -173,16 +173,35 @@ int BasisSet::n_ecp_core() const {
     return ncoreelectrons;
 }
 
+int BasisSet::atom_to_period(int Z) {
+    if (Z > 118) {
+        throw PSIEXCEPTION("Atomic number beyond Oganesson");
+    }
+    static const std::vector<int> full_shell_values = {0, 2, 10, 18, 36, 54, 86, 118};
+    auto period = std::lower_bound(full_shell_values.begin(), full_shell_values.end(), Z) - full_shell_values.begin();
+    return period;
+}
+
+int BasisSet::period_to_full_shell(int p) {
+    if (p > 7) {
+        throw PSIEXCEPTION("Atomic number beyond Oganesson");
+    }
+    static const std::vector<int> full_shell_values = {0, 2, 10, 18, 36, 54, 86, 118};
```

Comment:
Since full_shell_values is already a ```static const``` array, you could declare it outside the functions so you avoid having it in two places.